internal/services/databricks/databricks_virtual_network_peering_resource.go

```diff
@@ -0,0 +1,360 @@
+			databricksVirtualNetworkId = *model.Properties.DatabricksVirtualNetwork.Id
+		}
+		d.Set("virtual_network_id", databricksVirtualNetworkId)
+
+		remoteVirtualNetworkId := ""
+		if model.Properties.RemoteVirtualNetwork.Id != nil {
+			remoteVirtualNetworkId = *model.Properties.RemoteVirtualNetwork.Id
+		}
+		d.Set("remote_virtual_network_id", remoteVirtualNetworkId)
+	}
+
+	return nil
+}
+
+func resourceDatabricksVirtualNetworkPeeringUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataBricks.VnetPeeringClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for Azure ARM databricks virtual network peering update.")
+
+	id, err := vnetpeering.ParseVirtualNetworkPeeringID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	existing, err := client.Get(ctx, *id)
+	if err != nil {
+		return fmt.Errorf("retrieving Databricks %s: %+v", *id, err)
+	}
+
+	// these are the only updatable values, so everything else in the existing.Model should still be unchanged...
+	if d.HasChange("allow_forwarded_traffic") {
+		existing.Model.Properties.AllowForwardedTraffic = pointer.To(d.Get("allow_forwarded_traffic").(bool))
+	}
+
+	if d.HasChange("allow_gateway_transit") {
+		existing.Model.Properties.AllowGatewayTransit = pointer.To(d.Get("allow_gateway_transit").(bool))
+	}
+
+	if d.HasChange("allow_virtual_network_access") {
+		existing.Model.Properties.AllowVirtualNetworkAccess = pointer.To(d.Get("allow_virtual_network_access").(bool))
+	}
+
+	if d.HasChange("use_remote_gateways") {
+		existing.Model.Properties.UseRemoteGateways = pointer.To(d.Get("use_remote_gateways").(bool))
+	}
+
+	locks.ByID(databricksVnetPeeringsResourceType)
+	defer locks.UnlockByID(databricksVnetPeeringsResourceType)
+
+	future, err := client.CreateOrUpdate(ctx, *id, *existing.Model)
+	if err != nil {
+		return fmt.Errorf("updating Databricks %s: %+v", *id, err)
+	}
+
+	if err := future.Poller.PollUntilDone(ctx); err != nil {
+		return fmt.Errorf("waiting for update of Databricks %s: %+v", *id, err)
+	}
+
+	return resourceDatabricksVirtualNetworkPeeringRead(d, meta)
+}
+
+func resourceDatabricksVirtualNetworkPeeringDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataBricks.VnetPeeringClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := vnetpeering.ParseVirtualNetworkPeeringID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// Block all changes to any resource of this type...
+	locks.ByID(databricksVnetPeeringsResourceType)
+	defer locks.UnlockByID(databricksVnetPeeringsResourceType)
+
+	future, err := client.Delete(ctx, *id)
+	if err != nil {
+		return fmt.Errorf("deleting Databricks %s: %+v", *id, err)
+	}
+
+	if err = future.Poller.PollUntilDone(ctx); err != nil {
+		return fmt.Errorf("waiting for deletion of Databricks %s: %+v", *id, err)
+	}
```

Comment:
Can this be reduced to a single call to `client.DeleteThenPoll()`?

Reply:
@manicminer, I noticed that too in the debug, this was calling delete twice... Fixed.